### DISCUSSION TITLE
导出为手机长截图

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2612] [插件说明]
描述
---

> 将当前的页面导出为适合手机阅读的截图

地址
---

[https://simpread.ksria.cn/plugins/details/1MarRdpJum](https://simpread.ksria.cn/plugins/details/1MarRdpJum)

功能
---

1. 快捷键 `s c` 转换

2. 三种尺寸：

   - 大图（默认）：适合各种手机设备，宽度为 `650px`
   - 正常：适合 iPhone 12 Pro Max，宽度为 `428px`
   - 小图：适合 iPhone 12 mini，宽度为 `360px`
 
3. 支持阅读模式的主题色

4. 支持标注

5. 导出的截图包含当前 URL 的二维码

选项
---

![image](https://user-images.githubusercontent.com/81074/127732333-abffb90a-c2f4-4a74-948e-9a313df441fb.png)

导出效果
---

因图较大，[点击查看](https://user-images.githubusercontent.com/81074/127732363-6cfd6971-2793-4ace-88b7-21953e713d5d.png)

更新日志
---

### 1.0.2

- `Optimize` 为配合 [插件管理器](https://github.com/Kenshin/simpread/discussions/4221) 的更新方式做了数据结构的优化。

### 1.0.1

- `Optimize` 快捷键 `s c` 更改为 `m s` （因为 s c 现在是分享卡的快捷键

### 1.0.0

1. 快捷键 `s c` 转换

2. 三种尺寸：

   - 大图（默认）：适合各种手机设备，宽度为 `650px`
   - 正常：适合 iPhone 12 Pro Max，宽度为 `428px`
   - 小图：适合 iPhone 12 mini，宽度为 `360px`
 
3. 支持阅读模式的主题色

4. 支持标注

5. 导出的截图包含当前 URL 的二维码

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by wyl772191140** [comment]
快键键sc命令，使用不了，按了没反应

**Reply by Kenshin** [reply]
请告诉具体 URL

**Reply by sailcom** [reply]
我也是同样的问题，在页面中无法使用sc快捷键调出截图窗口（其他插件如无图版的快捷键可正常使用）；
尝试了sspai.com、news.163.com等多个页面均无法使用；
环境：win10+edge95.0+简悦2.2.0.520

**Reply by Kenshin** [reply]
> 我也是同样的问题，在页面中无法使用sc快捷键调出截图窗口（其他插件如无图版的快捷键可正常使用）； 尝试了sspai.com、news.163.com等多个页面均无法使用； 环境：win10+edge95.0+简悦2.2.0.520

emmm 请问在插件触发器中能否看到它？或者在**隐身模式下仅使用简悦的方式** 看看是不是其它的扩展干扰造成的？

**Reply by sailcom** [reply]
> emmm 请问在插件触发器中能否看到它？或者在**隐身模式下仅使用简悦的方式** 看看是不是其它的扩展干扰造成的？

- 在插件触发器中可以看到该插件，点击后也可以正常使用。仅仅是无法使用sc快捷键调出该功能。
- 尝试了隐身模式（InPrivate窗口），只允许简悦插件，且只安装“导出为手机长截图”一个插件时，依旧无法通过sc快捷键调出。
- 尝试更改搜狗输入法的中英文状态，也无法用sc调出。（通过插件触发器点击可以正常使用）
- 高级设定-快捷键-“禁用全部快捷键”选项已关闭，允许使用快捷键进行操作（其他插件，比如无图版的快捷键可以正常使用）

**Reply by Kenshin** [reply]
@sailcom 

嗨，谢谢这么细致的测试，我先记录 Bug 看看能否修复这个问题，你可以使用下面的手动方式

![image](https://user-images.githubusercontent.com/81074/141400550-a6728803-e21d-4ed2-81ed-32f8e4b5e914.png)

**Comment by taysxiao** [comment]
导出时显示开始转换，然后弹出转换失败！

**Reply by Kenshin** [reply]
嗨，请提供具体 URL 并新建一个 https://github.com/kenshin/simpread/issues/new

PS：长截图确实在某些网页无法使用，但因为我修改了算法，所以发生的概率很低了。

**Comment by taysxiao** [comment]
后来又试了下其他网页，基本都能用。第一次错误那个网页实在是太长了，可能你调用的哪个模块存在一个处理的上限。基本不影响日常的使用。谢谢。

Kenshin Wang ***@***.***> 于 2021年12月30日周四 下午4:57写道：

**Reply by sailcom** [reply]
请问您是什么浏览器？我至今还是不能用sc快捷键。

**Reply by Kenshin** [reply]
@sailcom 

Chrome （跟哪个浏览器没关系...

你可以这样试试：

> 导出配置文件（或者确保有之前的配置文件），然后重置（或重新安装）简悦。然后只安装这个插件，并在**隐身模式下仅使用简悦的方式** ，看看能否使用快捷键。

**Reply by sailcom** [reply]
> Chrome （跟哪个浏览器没关系...
> 你可以这样试试：
> > 导出配置文件（或者确保有之前的配置文件），然后重置（或重新安装）简悦。然后只安装这个插件，并在**隐身模式下仅使用简悦的方式** ，看看能否使用快捷键。

按照该提示又尝试了一遍，依然还是不能用快捷键（环境及步骤如 https://github.com/Kenshin/simpread/discussions/2612#discussioncomment-1629058
感觉可能跟电脑环境也有关系，到时候我换一台电脑重新尝试一下。
我暂时先用手动点击的方式吧。感谢Kenshin

**Reply by Kenshin** [reply]
@sailcom 

小事情，你可以试试简悦内置的其它快捷键是否可用？

**Reply by sailcom** [reply]
> 小事情，你可以试试简悦内置的其它快捷键是否可用？

其他快捷键都没问题，唯独这个 sc 无法使用

**Reply by HongFengLuoYe** [reply]
如何去掉底部的二维码和说明？

**Comment by karsten758** [comment]
导出部分网站有白色长方块挡住内容
![4](https://github.com/Kenshin/simpread/assets/10143989/875df1df-dacb-451a-af99-67491be4c8af)

**Reply by Kenshin** [reply]
1. 试试把目录隐藏？
2. **隐身模式下仅使用简悦的方式**，排除其它扩展的干扰。
3. 每个页面都有这个问题吗？

**Reply by karsten758** [reply]
1.使用的快捷键打开的插件，没有通过目录菜单进入插件
2.隐身模式同样出现这个问题
3.这个站点的每个页面基本会出现，其他站点不会出现
4.插件预览界面正常，转换后下载就出现上述长白块遮挡内容

**Reply by Kenshin** [reply]
> 这个站点的每个页面基本会出现，其他站点不会出现

如果只是个例的话，暂时没什么好办法，毕竟这种方式也有缺陷，在某些页面（尤其是有严苛防图的页面）使用效果不佳，虽然是个别情况，但毕竟存在，你可以使用一些第三方截图的 Chrome 扩展暂时弥补。

方便的话，可以提交这个问题 https://github.com/kenshin/simpread/issues/new
